apis/Google.Cloud.Diagnostics.Common/Google.Cloud.Diagnostics.Common/Trace/DelegatingTracer.cs

```diff
@@ -46,8 +46,8 @@ internal DelegatingTracer(Func<IManagedTracer> managedTracerGetter)
             _managedTracerGetter().RunInSpan(func, name, options);
 
         /// <inheritdoc />
-        public Task<T> RunInSpanAsync<T>(Func<Task<T>> func, string name, StartSpanOptions options = null) =>
-            _managedTracerGetter().RunInSpan(func, name, options);
+        public async Task<T> RunInSpanAsync<T>(Func<Task<T>> func, string name, StartSpanOptions options = null) =>
+            await _managedTracerGetter().RunInSpanAsync(func, name, options).ConfigureAwait(false);
```

Comment:
Just return the task from `RunInSpanAsync`: remove `async`, `await`, and `.ConfigureAwait(false)`. No need for the compiler generated async machinery here.

Reply:
I think the `ConfigureAwait(false)` should stay since this code might run in environments with a synchronization context.

Reply:
I've removed async/await from method body. I guess it will be just passing task and without await it won't capture context. However, if I'm mistaken, please correct me.

Reply:
You shouldn't need `ConfigureAwait(false)` because you're not awaiting anything anymore. You're passing the Task back to the caller to decided when to wait for it, so the caller that ultimately awaits on it decides whether to call ConfigureAwait or not.